examples/models/ResNet50.cpp

```diff
@@ -0,0 +1,1293 @@
+    }
+    /* -------------- conv -> relu -> conv -> relu -> pool -------------- */
+    auto conv1_src_md = memory::desc(conv1_src_dims, dt::f32, mem_format);
+    auto conv1_src_memory = memory(conv1_src_md, eng);
+    write_to_dnnl_memory(src_data_copied.data(), conv1_src_memory);
+    //     std::vector<float> test_data;
+    //     test_data.resize(product(conv1_src_dims));
+    //     read_from_dnnl_memory(test_data.data(), conv1_src_memory);
+    //     std::vector<float> padded_data;
+    //     memory::dims pad_r {3, 3};
+    //     auto conv1_src_md_padded = do_input_padding(
+    //             conv1_src_memory, padded_data, conv1_padding, pad_r);
+    //     auto conv1_src_mem_padded = memory(conv1_src_md_padded, eng);
+    //     conv1_padding = {0, 0};
+    //     write_to_dnnl_memory(padded_data.data(), conv1_src_mem_padded);
+    auto conv1_dst_md = memory::desc(conv1_dst_dims, dt::f32, mem_format);
+    auto conv1_dst_memory = memory(conv1_dst_md, eng);
+    STAMP_OUT_WEIGHTS_CONV(1, conv1_weights_dims, conv1_bias_dims);
+    STAMP_OUT_BNORM(1, OC);
+    do_conv(eng, net, net_args, weights_1.data(), biases_1.data(),
+            conv1_weights_dims, conv1_bias_dims, conv1_strides, conv1_padding,
+            conv1_src_memory, conv1_dst_memory, false);
+    do_bnorm(eng, net, net_args, mean_1.data(), var_1.data(),
+            scale_shift_1.data(), OC, conv1_dst_memory);
+    do_relu(eng, net, net_args, conv1_dst_memory);
+
+    /* Pooling dims */
+    const memory::dim KH = 3, // kernel height
+            KW = 3, // kernel width
+            PH_L = 1, // height padding: left
+            PH_R = 1, // height padding: right
+            PW_L = 1, // width padding: left
+            PW_R = 1, // width padding: right
+            SH = 2, // height-wise stride
+            SW = 2; // width-wise stride
+    WIDTH = WIDTH / 2, HEIGHT = HEIGHT / 2;
+    memory::dims kernel_dims = {KH, KW};
+    memory::dims strides_dims = {SH, SW};
+    memory::dims padding_dims_l = {PH_L, PW_L};
+    memory::dims padding_dims_r = {PH_R, PW_R};
+    memory::dims pool1_src_dims = {BATCH, OC, WIDTH, HEIGHT};
+    memory::dims pool1_dst_dims = {BATCH, OC, WIDTH / 2, HEIGHT / 2};
+    auto pool1_dst_md = memory::desc(pool1_dst_dims, dt::f32, mem_format);
+    auto pool1_dst_memory = memory(pool1_dst_md, eng);
+    do_pool(eng, net, net_args, strides_dims, kernel_dims, padding_dims_l,
+            padding_dims_r, conv1_dst_memory, pool1_dst_memory,
+            algorithm::pooling_max, false);
+    /* ---------------------------------- */
+
+    /* -----------------res2a_branch1----------------- */
+    /* -----------------conv > bnorm > scale----------------- */
+    block_params block_2a1_params {2, 1, 1, 0, false, false, BATCH, 64, 256, 56,
+            56, 1, 1, {0, 0}, {1, 1}};
+    res_net_block block_2a1(
+            eng, net, net_args, pool1_dst_memory, block_2a1_params);
+    //     WIDTH = WIDTH / 2, HEIGHT = HEIGHT / 2;
```

Comment:
Same here? The 2 defined macros `STAMP_OUT_WEIGHTS_CONV_BRANCHED` and `STAMP_OUT_BNORM_BRANCHED` could be removed if this code is deleted.

Reply:
Removed them.